src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/services/storage/OcrValidationRetryManager.java

```diff
@@ -0,0 +1,121 @@
+package uk.gov.hmcts.reform.bulkscanprocessor.services.storage;
+
+import com.azure.core.util.Context;
+import com.azure.storage.blob.BlobClient;
+import com.azure.storage.blob.models.BlobRequestConditions;
+import org.apache.commons.lang3.StringUtils;
+import org.slf4j.Logger;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Component;
+
+import java.time.LocalDateTime;
+import java.util.Map;
+
+import static java.time.LocalDateTime.now;
+import static java.time.LocalDateTime.parse;
+import static org.slf4j.LoggerFactory.getLogger;
+import static uk.gov.hmcts.reform.bulkscanprocessor.util.TimeZones.EUROPE_LONDON_ZONE_ID;
+
+@Component
+public class OcrValidationRetryManager {
+
+    private static final Logger logger = getLogger(OcrValidationRetryManager.class);
+
+    private static final String RETRY_COUNT_METADATA_PROPERTY = "ocrValidationRetryCount";
+    private static final String RETRY_DELAY_EXPIRATION_TIME_METADATA_PROPERTY =
+        "ocrValidationRetryDelayExpirationTime";
+
+    private final LeaseAcquirer leaseAcquirer;
+    private final int ocrValidationMaxRetries;
+    private final int ocrValidationRetryDelaySec;
+
+    public OcrValidationRetryManager(
+        LeaseAcquirer leaseAcquirer,
+        @Value("${ocr-validation-max-retries}") int ocrValidationMaxRetries,
+        @Value("${ocr-validation-delay-retry-sec}") int ocrValidationRetryDelaySec
+    ) {
+        this.leaseAcquirer = leaseAcquirer;
+        this.ocrValidationMaxRetries = ocrValidationMaxRetries;
+        this.ocrValidationRetryDelaySec = ocrValidationRetryDelaySec;
+    }
+
+    public boolean isRetryDelayNotSetOrExpired(BlobClient blobClient) {
```

Comment:
Quite a mouthful.
Does this basically mean the passed argument is ready for retry?

Reply:
Renamed to `isReadyToRetry`